src/Data/Functor/Linear.hs

```diff
@@ -3,17 +3,22 @@
 
 -- | = The data functor hierarchy
 --
--- This module defines the data functor library. These are linear functors which
--- are better understood as containers of data. Unlike unrestricted functor,
--- there is a split between such data functors and control functors which
--- represent effects (see "Control.Monad.Linear" for more).
+-- This module defines the data functor library. Unlike in the case of
+-- non-linear, unrestricted, functors, there is a split between data functors,
+-- which represent containers, and control functors which represent effects.
+-- Please read this
+-- [blog post](https://www.tweag.io/posts/2020-01-16-data-vs-control.html).
+-- For more details, see "Control.Monad.Linear".
+--
+-- * Linear data functors should be thought of as containers of data.
+-- * Linear data applicative functors should be thought of as containers
+-- that can be zipped.
+-- * Linear data traversible functors should be thought of as
+-- containers which store a finite number of values.
```

Comment:
Contrary to the equivalent bullet-point list in `Control.Monad`, I quite like this one!